File: app/src/test/java/net/sourcefusion/agiletools/sprintcalendar/BasicHolidayProviderTest.java
package net.sourcefusion.agiletools.sprintcalendar;


import org.joda.time.DateTime;
import org.joda.time.LocalDate;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class BasicHolidayProviderTest {

    @Test
    public void detectsHoliday() {
        BasicHolidayProvider basicHolidayProvider = new BasicHolidayProvider(new LocalDate(2016, 1, 6));

        assertThat(basicHolidayProvider.isHoliday(new LocalDate(2016, 1, 6)), is(true));
        assertThat(basicHolidayProvider.isWorkingDay(new LocalDate(2016, 1, 6)), is(false));
        assertThat(basicHolidayProvider.isHoliday(new LocalDate(2016, 1, 5)), is(false));
        assertThat(basicHolidayProvider.isWorkingDay(new LocalDate(2016, 1, 5)), is(true));
    }

    @Test
    public void detectsHolidayIgnoringTime() {
        BasicHolidayProvider basicHolidayProvider = new BasicHolidayProvider(new DateTime(2016, 1, 6, 12, 1).toLocalDate());

        assertThat(basicHolidayProvider.isHoliday(new DateTime(2016, 1, 6, 10, 25).toLocalDate()), is(true));
    }

    @Test
    public void detectsWeekends() {
        BasicHolidayProvider basicHolidayProvider = new BasicHolidayProvider();

        assertThat(basicHolidayProvider.isWeekend(new LocalDate(2016, 1, 9)), is(true));
        assertThat(basicHolidayProvider.isWeekend(new LocalDate(2016, 1, 10)), is(true));
    }

    @Test
    public void treatsWeekandsAsHolidays() {
        BasicHolidayProvider basicHolidayProvider = new BasicHolidayProvider();

        assertThat(basicHolidayProvider.isHoliday(new LocalDate(2016, 1, 9)), is(true));
        assertThat(basicHolidayProvider.isHoliday(new LocalDate(2016, 1, 10)), is(true));
    }

    @Test
    public void doesNotTreatsHolidayAsWeekend() {
        BasicHolidayProvider basicHolidayProvider = new BasicHolidayProvider(new LocalDate(2016, 1, 6));

        assertThat(basicHolidayProvider.isHoliday(new LocalDate(2016, 1, 6)), is(true));
        assertThat(basicHolidayProvider.isWeekend(new LocalDate(2016, 1, 6)), is(false));
    }
}

File: app/src/test/java/net/sourcefusion/agiletools/sprintcalendar/BasicHolidayProviderTest.kt
package net.sourcefusion.agiletools.sprintcalendar


import org.joda.time.DateTime
import org.joda.time.LocalDate
import org.junit.Test

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat

class BasicHolidayProviderTest {

    @Test
    fun detectsHoliday() {
        val basicHolidayProvider = BasicHolidayProvider(LocalDate(2016, 1, 6))

        assertThat(basicHolidayProvider.isHoliday(LocalDate(2016, 1, 6)), `is`(true))
        assertThat(basicHolidayProvider.isWorkingDay(LocalDate(2016, 1, 6)), `is`(false))
        assertThat(basicHolidayProvider.isHoliday(LocalDate(2016, 1, 5)), `is`(false))
        assertThat(basicHolidayProvider.isWorkingDay(LocalDate(2016, 1, 5)), `is`(true))
    }

    @Test
    fun detectsHolidayIgnoringTime() {
        val basicHolidayProvider = BasicHolidayProvider(DateTime(2016, 1, 6, 12, 1).toLocalDate())

        assertThat(basicHolidayProvider.isHoliday(DateTime(2016, 1, 6, 10, 25).toLocalDate()), `is`(true))
    }

    @Test
    fun detectsWeekends() {
        val basicHolidayProvider = BasicHolidayProvider()

        assertThat(basicHolidayProvider.isWeekend(LocalDate(2016, 1, 9)), `is`(true))
        assertThat(basicHolidayProvider.isWeekend(LocalDate(2016, 1, 10)), `is`(true))
    }

    @Test
    fun treatsWeekandsAsHolidays() {
        val basicHolidayProvider = BasicHolidayProvider()

        assertThat(basicHolidayProvider.isHoliday(LocalDate(2016, 1, 9)), `is`(true))
        assertThat(basicHolidayProvider.isHoliday(LocalDate(2016, 1, 10)), `is`(true))
    }

    @Test
    fun doesNotTreatsHolidayAsWeekend() {
        val basicHolidayProvider = BasicHolidayProvider(LocalDate(2016, 1, 6))

        assertThat(basicHolidayProvider.isHoliday(LocalDate(2016, 1, 6)), `is`(true))
        assertThat(basicHolidayProvider.isWeekend(LocalDate(2016, 1, 6)), `is`(false))
    }
}